test/taskcluster_secrets_api_test.js

```diff
@@ -0,0 +1,150 @@
+suite("TaskCluster-Secrets", () => {
+  var helper = require('./helper');
+  var assert = require('assert');
+  var slugid = require('slugid');
+
+  let testValueExpires  = {secret: {data: "bar"}, expires: "2066-10-06T07:25:54.957Z"};
+  let testValueExpires2 = {secret: {data: "foo"}, expires: "2066-10-06T07:25:54.957Z"};
+  let testValueExpired  = {secret: {data: "bar"}, expires: "2011-04-01T00:00:00.000Z"};
+
+  const FOO_KEY = slugid.v4();
+  const BAR_KEY = slugid.v4();
+
+  let testData = [
+    // The "Captain" clients
+    {
+      testName:   "Captain, write allowed key",
+      clientName: "captain-write",
+      apiCall:    "set",
+      name:        "captain:" + FOO_KEY,
+      args:       testValueExpires,
+      res:        {}
+    },
+    {
+      testName:   "Captain, write disallowed key",
+      clientName: "captain-write",
+      apiCall:    "set",
+      name:       "tennille:" + FOO_KEY,
+      args:       testValueExpires,
+      statusCode: 401 // It's not authorized!
+    },
+    {
+      testName:   "Captain (write only), fail to read.",
+      clientName: "captain-write",
+      apiCall:    "get",
+      name:        "captain:" + FOO_KEY,
+      statusCode: 401 // it's not authorized!
+    },
+    {
+      testName:   "Captain (write only), fail to write existing.",
+      clientName: "captain-write",
+      apiCall:    "set",
+      args:       testValueExpires2,
+      name:       "captain:" + FOO_KEY,
+      statusCode: 409 // it's not authorized!
```

Comment:
nit, but the comment "// it's not authorized!" is wrong, right?

You mean: "// conflict" I hope...